### PR TITLE
Feature/add pipeline variables

### PR DESCRIPTION
**Pull request by ionutleca**

### CODE REVIEW

File: .gitignore
Comment (anweiss) [inline .gitignore]: @ionutleca was this intended? best not to commit `node_modules/`

**Comment by anweiss** [comment]
@nishubansal @N-Usha this is a much needed feature for this Action. Can we get this reviewed and merged in once the `node_modules/` dir is removed from this PR?

**Comment by msoute** [comment]
@ionutleca is there any progress on the PR ? This feature would be very usefull

**Comment by mfcollins3** [comment]
So it looks like this PR is stuck in limbo. Is there some other way of getting this functionality into here? Being able to set the values of pipeline variables is really important. Can the core team just implement the functionality without using this PR? Or can I submit a PR with a similar code to get this moving?

**Comment by ionutleca** [comment]
I will try to get my current employer approval for the CLA required for this PR. If I find out that it takes more than 24h, I'll let you know if you should just create the PR with these changes yourself, @mfcollins3.

**Comment by mfcollins3** [comment]
Thanks, @ionutleca. I greatly appreciate you getting the CLA approved. I can't wait to start using this!

**Comment by FlipABit** [comment]
@ionutleca I'm assuming since 24h turned into 2+ months, one of us can go ahead and create the PR with these changes ourselves?

**Comment by ionutleca** [comment]
The CLA was approved a long time ago.

**Comment by bm77525-kr** [comment]
Apologies, the timeline was a bit confusing. It’s a shame because this is super useful, but they seemed to let it fall by the wayside.

**Comment by ionutleca** [comment]
No worries. It seems useful to me as well, but we might be the only ones :))

**Comment by bm77525-kr** [comment]
@nishubansal @N-Usha @anweiss Can anyone finally get this merged in?

**Comment by utricularia** [comment]
Any chance either this or the PR for adding build parameters might get merged in?

**Comment by JamesBurnside** [comment]
+1 to getting this merged in

File: src/pipeline.runner.ts
Comment (JamesBurnside) [inline src/pipeline.runner.ts:171]: Should this be using the `variables` variable from above instead of reparsing the variable string here?

```suggestion
            variables: variables
```

File: src/pipeline.runner.ts
Comment (JamesBurnside) [inline src/pipeline.runner.ts:132]: This will cause an unhandled exception when `this.taskParameters.azurePipelineVariables` is null or undefined. As this is an optional param best to wrap this in a conditional, e.g.:
```
        let variables = undefined;
        if (this.taskParameters.azurePipelineVariables) {
            variables = JSON.parse(this.taskParameters.azurePipelineVariables);
            Object.keys(variables).map(function (key, index) {
                let oldValue = variables[key]
                variables[key] = { value: oldValue }
            });
        }
```

Comment (ionutleca) [reply src/pipeline.runner.ts]: Thanks, @JamesBurnside. I rebased on the upstream and preferred to squash/fpush the original changes. I kept your suggestions as a separate commit though.

File: lib/pipeline.runner.js
Comment (JamesBurnside) [inline lib/pipeline.runner.js:136]: Looks like the js needs rebuilt here to include your latest change regarding safer use of the JSON.parse

File: src/pipeline.runner.ts
Comment (JamesBurnside) [reply src/pipeline.runner.ts]: Thanks for making these changes to such an outstanding PR (in both meanings of outstanding 😄). I've pinged the original owners of this action on our internal communication at Microsoft so hopefully we'll see some traction on getting this checking in

**Review by JamesBurnside — APPROVED** [review]
Tested this and using this solution in a fork of this repo for triggering an ADO release pipeline with variables that needed set. Approving but one more small change needed where the js file needs rebuilt to include the latest change 👍

File: lib/pipeline.runner.js
Comment (ionutleca) [reply lib/pipeline.runner.js]: yeah

**Pull request merged**
